firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteRemoteDocumentCache.java

```diff
@@ -229,23 +226,9 @@ public MutableDocument get(DocumentKey documentKey) {
   }
 
   @Override
-  public ImmutableSortedMap<DocumentKey, MutableDocument> getAllDocumentsMatchingQuery(
-      final Query query, IndexOffset offset) {
-    hardAssert(
-        !query.isCollectionGroupQuery(),
-        "CollectionGroup queries should be handled in LocalDocumentsView");
-
-    Map<DocumentKey, MutableDocument> allDocuments =
-        getAll(Collections.singletonList(query.getPath()), offset, Integer.MAX_VALUE);
-
-    ImmutableSortedMap<DocumentKey, MutableDocument> matchingDocuments =
-        DocumentCollections.emptyMutableDocumentMap();
-    for (MutableDocument document : allDocuments.values()) {
-      if (document.isFoundDocument() && query.matches(document)) {
-        matchingDocuments = matchingDocuments.insert(document.getKey(), document);
-      }
-    }
-    return matchingDocuments;
+  public Map<DocumentKey, MutableDocument> getAll(
+      final ResourcePath collection, IndexOffset offset) {
```

Comment:
nit: `final` on the `ResourcePath` argument isn't strictly required anymore. Can it be removed?

Reply:
Removed (now it fits into one line!)